### PR TITLE
Dala-1691_manual_maintenance_sprint43

### PR DESCRIPTION
**Pull request by ahoecherl**
# Manual Maintenance Sprint [NR]

Note: To create a PR using this template add the query parameter `template=manual_maintenance.md` to the merge request
creation URL (or simply copy this md file into the description)

# Maintenance tasks (to be completed by the assignee)

### Skipped updates

The following known issues need to be reviewed in case a compatible version is available. Add new known issues as they
appear.

- [ ] Update to gradle 8.0.2 is skipped due to task dependency issues in the building process. Backlog: DALA-1295
- [x] Update keycloak 21.0.1 is skipped due to issues in the realm.json, the new docker image, account console
- [x] Update "@vue/tsconfig" to >=0.2.0 introduces major changes in typescript rules (~500 TS Errors throughout the
  project and unresolved imports that are hard to fix), skipped.

### Gradle update

- [x] Execute `gradlew dependencyUpdates` to get a report on Dependencies with updates
- [x] Update `settings.gradle.kts` (for libraries), `build.gradle.kts` (for plugins) and `gradle.properties` (for jacoco
  and ktlint)
- [x] update the gradle wrapper: execute `gradle wrapper --gradle-version X.Y.Z`

### Dataland frontend

- [x] Update node version in `dataland-frontend/build.gradle.kts`
- [x] Update node packages: run the `updatepackages` script, e.g. by  `npm run updatepackages` to update versions in
  package.json
- [x] Run the `updatepackagelock`, e.g. by  `npm run updatepackagelock` script to update `package-lock.json` and check
  for security issues

### Dataland keycloak theme

- [x] Update node version in `dataland-keycloak/dataland_theme/login/build.gradle.kts`
- [x] Update node packages: run the `updatepackages` script, e.g. by  `npm run updatepackages` to update versions in
  package.json
- [x] Run the `updatepackagelock`, e.g. by  `npm run updatepackagelock` script to update `package-lock.json` and check
  for security issues

### Dockerfile updates

Update versions in the following dockerfiles

- [x] `./dataland-api-key-manager/Dockerfile`
- [x] `./dataland-api-key-manager/DockerfileBase`
- [x] `./dataland-api-key-manager/DockerfileTest`
- [x] `./dataland-backend/Dockerfile`
- [x] `./dataland-backend/DockerfileBase`
- [x] `./dataland-backend/DockerfileTest`
- [x] `./dataland-csvconverter/Dockerfile`
- [x] `./dataland-document-manager/Dockerfile`
- [x] `./dataland-document-manager/DockerfileBase`
- [x] `./dataland-document-manager/DockerfileTest`
- [x] `./dataland-e2etests/Dockerfile`
- [x] `./dataland-e2etests/DockerfileBase`
- [x] `./dataland-frontend/Dockerfile`
- [x] `./dataland-frontend/DockerfileTest`
- [x] `./dataland-internal-storage/Dockerfile`
- [x] `./dataland-internal-storage/DockerfileBase`
- [x] `./dataland-internal-storage/DockerfileTest`
- [x] `./dataland-qa-service/Dockerfile`
- [x] `./dataland-qa-service/DockerfileBase`
- [x] `./dataland-qa-service/DockerfileTest`
- [x] `./dataland-rabbitmq/Dockerfile`
- [x] `./dataland-inbound-admin-proxy/Dockerfile`
- [ ] `./dataland-inbound-proxy/DockerfileBase`
- [x] `./dataland-inbound-proxy/Dockerfile`
- [x] `./dataland-pgadmin/Dockerfile`
- [x] `./dataland-keycloak/Dockerfile`  (also update realm json files with new version)
- [x] `./base-dockerfiles/DockerfileGradle`
- [x] Update the versions of the external images for api-key-manager-db, backend-db, keycloak-db, internal-storage-db,
  document-manager-db and frontend-dev
- [x] Check if there are any services in the `docker-compose.yml` file that have not gotten an update yet (e.g. a new
  service that is not covered by the tasks above)

## Dataland Monitoring

- [x] Go to the monitoring repository and execute the tasks described in the manual maintenance template there

## Server updates

Execute `sudo apt-get update && sudo apt-get upgrade` on

- [x] dev1.dataland.com
- [ ] dev2.dataland.com
- [x] test.dataland.com
- [ ] tunnel.dataland.com
- [ ] letsencrypt.dataland.com
- [ ] monitoring.dataland.com
- [ ] (OPT) dataland.com

### ssh-keys maintenance

check that all ssh-keys are set and erased from people that have left

- [ ] dev1.dataland.com
- [ ] dev2.dataland.com
- [x] test.dataland.com
- [ ] tunnel.dataland.com
- [ ] letsencrypt.dataland.com
- [ ] monitoring.dataland.com
- [ ] (OPT) dataland.com

## Check keycloak automatic logout if inactive

- [x] Check that you are automatically logged out if you are idle for too long and also get notified about this by a
  pop-up. Timeouts are defined as `TIME_BEFORE_REFRESH_TOKEN_EXPIRY_TO_DISPLAY_SESSION_WARNING_IN_MS`
  in `./dataland-frontend/src/utils/Constants.ts` and `ssoSessionIdleTimeout`
  in `./dataland-keycloak/realms/datalandsecurity-realm.json`.

## Check e-mails sent by keycloak

- [ ] Check that the verification e-mail is sent and displayed correctly
- [ ] Check that the reset password e-mail is sent and displayed correctly
- [ ] Check that the password can be reset
- [ ] Check that account linking via e-mail verification works correctly
- [ ] Check that account linking via username & password verification works correctly

## Check e-mails sent by backend

- [ ] Send an invitation (data) request from one of the dev servers and check if the e-mail to info@dataland.com
  contains the right attachments and is displayed correctly.

## Check RabbitMQ dead letter queue and disk space

- [ ] RabbitMQ does need at least 768MB of free disk space to operate. `ssh` into all servers and check the available
  disk space with `df` command. If the open disk space is close to the minimum requirement, clear up disk space
  with `docker image prune`.
- [ ] On all environments, no new messages should have been added to the dead letter queue since the last manual
  maintenance. If new messages have appeared this does need to be investigated. The dead letter queue can be accessed
  and messages on it read in the RabbitMQ GUI. Access it by port-forwarding port `6789` from the server and then
  accessing the GUI at `localhost:6789/rabbitmq`. After login, the dead letter queue can be found at Queues &rarr;
  deadLetterQueue &rarr; Get message.

## Conclusion

- [ ] After updating all components check if everything still works
- [ ] The new version is deployed to a dev server using this branch and real data
    - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
    - [ ] It's verified that real data has been used
    - [ ] It's verified that everything seems to be working fine by manually using the website
- [ ] This template has been updated to reflect the latest state of tasks required and known issues with upgrades
- [ ] The Merge Request commit message needs to contain 'manual maintenance' to satisfy the CI maintenance check in
  future commits

# Review (to be completed by the reviewer)

- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github.
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described above
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed with fake data to the dev server "dev1" using this branch
  - [x] Run with setting `Use real data` turned off and `Reset non-user related Docker Volumes & Re-populate` turned on
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] It's verified that everything seems to be working fine by manually using the website
  - [x] All implemented Social Logins have been tested manually in the UI
  - [x] Go to the swagger-UI, authorize, run a "GET" request to the companies endpoint and assure that your
    authorization has worked by assuring that you get a 200 response
- [x] The new version is deployed with real data to a dev server using this branch
  - [x] Run with both settings `Reset non-user related Docker Volumes & Re-populate` and `Use real data` turned on
  - [x] It's verified that the CD run is green
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma
- [x] The local Dev stack still works: execute `startDevelopmentStack.sh` and execute `npm run testpipeline`
  and `npm run testcomponent` in dataland-frontend
- [ ] After(!) the cypress tests have passed locally, execute the backend-e2e-tests `./gradlew dataland-e2etests:test`
- [x] Locally: Go to the swagger-UI, authorize, run a "GET" request to the companies endpoint and assure that your
  authorization has worked by assuring that you get a 200 response

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md [hunk 1]
```diff
@@ -12,9 +12,10 @@ creation URL (or simply copy this md file into the description)
 The following known issues need to be reviewed in case a compatible version is available. Add new known issues as they
 appear.
 
-- [ ] Update keycloak 21.0.1 is skipped due to issues in the realm.json, the new docker image, account console
```
Comment (ppuetsch) [inline .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md:15]: das muss wieder rein.

File: settings.gradle.kts [hunk 2]
```diff
@@ -44,23 +44,24 @@ dependencyResolutionManagement {
 
             library("slf4j-api", "org.slf4j:slf4j-api:2.0.7")
 
-            library("jackson-dataformat-csv", "com.fasterxml.jackson.dataformat:jackson-dataformat-csv:2.14.2")
-            library("jackson-databind", "com.fasterxml.jackson.core:jackson-databind:2.14.2")
-            library("jackson-datatype-jsr310", "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.2")
-            library("jackson-kotlin", "com.fasterxml.jackson.module:jackson-module-kotlin:2.14.2")
+            library("jackson-dataformat-csv", "com.fasterxml.jackson.dataformat:jackson-dataformat-csv:2.15.1")
+            library("jackson-databind", "com.fasterxml.jackson.core:jackson-databind:2.15.1")
```
Comment (ppuetsch) [inline settings.gradle.kts:48]: There is 2.15.2

File: build.gradle.kts [hunk 3]
```diff
@@ -38,25 +38,25 @@ tasks.dependencyUpdates.configure {
...
-    id("com.github.node-gradle.node") version "4.0.0" apply false
-    id("org.springframework.boot") version "3.0.6" apply false
+    id("com.github.jk1.dependency-license-report") version "2.3"
+    id("io.gitlab.arturbosch.detekt") version "1.23.0"
+    id("com.github.node-gradle.node") version "5.0.0" apply false
+    id("org.springframework.boot") version "3.1.0" apply false
     id("org.jlleitschuh.gradle.ktlint") version "11.3.2"
     kotlin("jvm") version "1.8.21"
     kotlin("plugin.spring") version "1.8.21" apply false
     id("org.sonarqube") version "4.0.0.2929"
```
Comment (ppuetsch) [inline build.gradle.kts:55]: 4.2.0.3129

File: settings.gradle.kts [hunk 4]
```diff
@@ -44,23 +44,24 @@ dependencyResolutionManagement {
...
             library("bcpkix-jdk15on", "org.bouncycastle:bcpkix-jdk15on:1.70")
             library("bcprov-jdk15on", "org.bouncycastle:bcprov-jdk15on:1.70")
 
-            library("cloudevents-api", "io.cloudevents:cloudevents-api:2.4.2")
-            library("cloudevents-json-jackson", "io.cloudevents:cloudevents-json-jackson:2.4.2")
-            library("cloudevents-core", "io.cloudevents:cloudevents-core:2.4.2")
+            library("cloudevents-api", "io.cloudevents:cloudevents-api:2.5.0")
+            library("cloudevents-json-jackson", "io.cloudevents:cloudevents-json-jackson:2.5.0")
+            library("cloudevents-core", "io.cloudevents:cloudevents-core:2.5.0")
             library("spring-cloud-function-context", "org.springframework.cloud:spring-cloud-function-context:4.0.2")
```
Comment (ppuetsch) [inline settings.gradle.kts:58]: 4.0.3

File: settings.gradle.kts [hunk 5]
```diff
@@ -44,23 +44,24 @@ dependencyResolutionManagement {
...
-            library("mailjet-client", "com.mailjet:mailjet-client:5.2.2")
+            library("mailjet-client", "com.mailjet:mailjet-client:5.2.3")
 
             library("pdfbox", "org.apache.pdfbox:pdfbox:2.0.28")
 
             library("mockito-kotlin", "org.mockito.kotlin:mockito-kotlin:4.1.0")
```
Comment (ppuetsch) [inline settings.gradle.kts:63]: 5.0.0

File: .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md (hunk 1 above)
Comment (ahoecherl) [reply .github/PULL_REQUEST_TEMPLATE/manual_maintenance.md:15]: Addressed

File: settings.gradle.kts (hunk 2 above)
Comment (ahoecherl) [reply settings.gradle.kts:48]: Addressed

File: build.gradle.kts [hunk 6]
```diff
@@ -38,25 +38,25 @@ tasks.dependencyUpdates.configure {
...
 }
 
 java.sourceCompatibility = JavaVersion.VERSION_17
 
 plugins {
-    id("com.github.jk1.dependency-license-report") version "2.1"
-    id("io.gitlab.arturbosch.detekt") version "1.22.0"
-    id("com.github.node-gradle.node") version "4.0.0" apply false
-    id("org.springframework.boot") version "3.0.6" apply false
+    id("com.github.jk1.dependency-license-report") version "2.3"
```
[the remark this answers is not in the document]
Comment (ahoecherl) [reply build.gradle.kts:48]: Addressed

File: build.gradle.kts (hunk 3 above)
Comment (ahoecherl) [reply build.gradle.kts:55]: Addressed

File: settings.gradle.kts (hunk 4 above)
Comment (ahoecherl) [reply settings.gradle.kts:58]: Addressed

File: settings.gradle.kts (hunk 5 above)
Comment (ahoecherl) [reply settings.gradle.kts:63]: This new major release very fresh of the press - 14 hours ago apparently. [Github Readme still states 4.1](https://github.com/mockito/mockito-kotlin)

Nevertheless, I've changed it to 5.0.0

File: build.gradle.kts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ppuetsch) [reply build.gradle.kts]: ok

File: build.gradle.kts (hunk 3 above)
Comment (ppuetsch) [reply build.gradle.kts]: ok

File: settings.gradle.kts (hunk 5 above)
Comment (ppuetsch) [reply settings.gradle.kts]: ok

File: settings.gradle.kts (hunk 2 above)
Comment (ppuetsch) [reply settings.gradle.kts]: well - all jacskon-thingys to 2.15.2

Comment (ppuetsch) [reply settings.gradle.kts]: ok

**Pull request merged**